source/kernel/src/comms/kchannel.rs

```diff
@@ -55,11 +55,34 @@ impl<T> Deref for KChannel<T> {
     }
 }
 
+fn right_size(mut cap: usize) -> usize {
+    if cap < 2 {
+        tracing::warn!(
```

Comment:
take it or leave it: what do you think about having this panic in debug mode, or at least with `cfg(test)`? that way, when we're running tests, it fails loudly, but if it somehow slips into a real build, we just fix it.

**Edit**: oh i see that we assert it elsewhere. that's fine!